sdk/spring/azure-spring-boot-starter-active-directory/README.md

```diff
@@ -39,22 +39,45 @@ The way to obtain group relationship that will determine which graph api will be
 * **direct**: the default value, get groups that the user is a direct member of. For details, see [list memberOf][graph-api-list-member-of] api.
 * **transitive**: Get groups that the user is a member of, and will also return all groups the user is a nested member of. For details, see [list transitive memberOf][graph-api-list-transitive-member-of] api.
 
-### Authenticate in frontend
-Sends bearer authorization code to backend, in backend a Spring Security filter `AADAuthenticationFilter` validates the Jwt token from Azure AD and save authentication. The Jwt token is also used to acquire a On-Behalf-Of token for Azure AD Graph API so that authenticated user's membership information is available for authorization of access of API resources. 
-Below is a diagram that shows the layers and typical flow for Single Page Application with Spring Boot web API backend that uses the filter for Authentication and Authorization.
-![Single Page Application + Spring Boot Web API + Azure AD](resource/auth-in-frontend-with-aad-filter.png)
+### Web Server
+Based on Azure AD as a Web Server, it uses OAuth2 authorization code flow to authentication, and authorizes resources based on the scopes or roles claim in the access token. 
+Provide a convenient way to quickly access other resource server, other resources should be registered as `ClientRegistration`, use `@RegisteredOAuth2AuthorizedClient` annotation to mark the client resource, Security will help automatically obtain valid access tokens based on the root refresh token, business methods will use the corresponding access token to request client resources.
 
+#### Standalone web server usage
+Only as a Web Server, no further access to other resources protected by Azure AD.
+![Standalone Web Server](resource/aad-based-standalone-web-server.png)
 
-### Authenticate in backend
-Auto configuration for common Azure Active Directory OAuth2 properties and `OAuth2UserService` to map authorities are provided.
+* Access restricted resources of Web Server, login with credentials using default scopes.
+* Return secured data.
+
+#### Web server access resources usage
+Web Server and resource server use scenarios, Web server access the resources of resource server which is protected by Azure AD.
+![Web Server Access Resources](resource/add-based-web-server-access-resources.png)
+
+* Login with credentials, the scope includes all other clients. 
+* Auto-refresh the access token of other clients based on the root refresh token.
+* Use each client's access token to request restricted resource.
+* Return secured data.
+
+### Resource Server
+Based on Azure AD as a Resource Server, it uses `BearerTokenAuthenticationFilter` authorize request. The current resource server also can access other resources, there's a similar method to the Web server usage to obtain access to the client access token, the difference is the access token obtained based on the `MSAL On-Behalf-Of` process.
```

Comment:
Do we have this `BearerTokenAuthenticationFilter`?

Reply:
Yes, as a resource server this filter will work.